trin-state/src/jsonrpc.rs

```diff
@@ -81,6 +82,26 @@ impl StateRequestHandler {
                     };
                     let _ = request.resp.send(response);
                 }
+                StateEndpoint::Offer => {
+                    let response = match OfferParams::try_from(request.params) {
+                        Ok(val) => {
+                            let content_keys =
+                                val.content_keys.iter().map(|x| x.to_vec()).collect();
```

Comment:
*nitpick* but i'd prefer using something like `key` rather than `x` here and in `trin-history`